python/pyspark/sql/session.py

```diff
@@ -483,13 +483,22 @@ def _inferSchema(self, rdd, samplingRatio=None, names=None):
                 row, names, infer_dict_as_struct=infer_dict_as_struct)).reduce(_merge_type)
         return schema
 
-    def _createFromRDD(self, rdd, schema, samplingRatio):
+    def _create_verified_converter(self, schema, verifySchema):
+        converter = _create_converter(schema)
+        verify_func = _make_type_verifier(schema) if verifySchema else lambda _: True
+
+        def verify_and_convert(obj):
+            verify_func(obj)
+            return converter(obj)
+        return verify_and_convert
```

Comment:
I'm a little confused by this, if `verifySchema` is `False` then could you just return `converter`?

Reply:
simplified now